doc/fgpu.design.rst

```diff
@@ -695,11 +695,18 @@ the fractional part. Even passing :math:`f` in single precision can lead
 to large errors.
 
 To avoid these problems, fixed-point computations are used. Phase is
-represented as a fractional number of cycles, scaled by :math:`2^{32}` and
-stored in a 32-bit integer. When performing arithmetic on values encoded this
+represented as a fractional number of cycles, scaled by :math:`2^{64}` and
+stored in a 64-bit integer. When performing arithmetic on values encoded this
 way, the values may overflow and wrap. The high bits that are lost represent
 complete cycles, and so have no effect on phase.
 
+This is sufficient to avoid any rounding errors inside the kernel, where
+:math:`t` is limited to the number of samples processed by a single kernel
+invocation. However, over much longer timescales (days) even quantising the
+frequency to 64-bit fixed point could lead to a small phase drift. The Python
```

Comment:
Just out of interest, Re: Longer timescales, is that a probable/likely use case/scenario?

Reply:
The mixer phase is based on the ADC timestamp, which is only reset every day or two with a global sync. It's 48-bit, so at the end of that time, the error will be on the order of 2^48 / 2^64 = 2^-16. That's probably not enough to actually worry about, but it's still much worse than fp32 accuracy.